libqpdf/QPDFObjectHandle.cc

```diff
@@ -2080,6 +2112,10 @@ QPDFObjectHandle::parseInternal(PointerHolder<InputSource> input,
                 object = QPDFObjectHandle(new QPDF_Array(olist));
                 setObjectDescriptionFromInput(
                     object, context, object_description, input, offset);
+                // The `offset` points to the next of "[".
+                // Set the rewind offset to point to the beginning of "[".
+                object.setParsedOffset(offset - 1);
```

Comment:
After reading a token, InputSource's getLastOffset() method always returns the offset of the beginning of the token. I wonder whether this could be used safely to create a cleaner way of handling this than "knowing" that you are just after the "[". When I see this code, I worry about whether the right thing happens if there is whitespace before or after the "[". If you can't use getLastOffset here because of some other reason in the code that I am forgetting, then I would want to double check that there is a test with different kinds of space around the brackets. Most likely there is already, but we should double check and make sure the comment here is specific about whitespace being handled propery.

Reply:
> After reading a token, InputSource's getLastOffset() method always returns the offset of the beginning of the token.

Unfortunately, `input->getLastOffset()` points the offset of "]" that is closing array instead of opening array.

> When I see this code, I worry about whether the right thing happens if there is whitespace before or after the "[".

`qpdf/qtest/qpdf/minimal.pdf` used in the tests has several whitespace pattern before and after the "[".

* both before and after the "["

```
  /Kids [
    3 0 R
  ]
```

* only before the "["

```
  /MediaBox [0 0 612 792]
```

* only after the "["

```
5 0 obj
[
  /PDF
```

These are working correctly.

Reply:
Great, thanks for the confirmation.